### PR TITLE
Code for blur filter

### PR DESCRIPTION
**Pull request by Tushar7-coder**

### CODE REVIEW

**Comment by chandrakant100** [comment]
Everything is good! Let's wait for @Himanshu40 reply.

File: helpers.cpp [hunk 1]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
 
 // Definition for Blur Filter
 void blur(const int &height, const int &width, RGBTRIPLE **image) {
-    // code to be written
+    // initialize variables to 0
+    int sumBlue {0};
+    int sumGreen {0};
+    int sumRed {0};
+    int count {0};
```
Comment (Himanshu40) [inline helpers.cpp:25]: Please align 0 perfectly.

File: helpers.cpp [hunk 2]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+        for (int j {0}; j < height; ++j) {
+        //Reinitialization of variables after each iteration
+            sumBlue  = 0;
+            sumGreen = 0;
+            sumRed   = 0;
+            count    = 0;
+
+            // sums values of the pixel and 8 neighboring ones, skips iteration if it goes outside the pic
+            for (int k {-1}; k < 2; ++k) {
+
```
Comment (Himanshu40) [inline helpers.cpp:39]:
```suggestion
```
Remove this unnecessary new line.

File: helpers.cpp [hunk 3]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+            sumBlue  = 0;
+            sumGreen = 0;
+            sumRed   = 0;
+            count    = 0;
+
+            // sums values of the pixel and 8 neighboring ones, skips iteration if it goes outside the pic
+            for (int k {-1}; k < 2; ++k) {
+            
+                if (j + k < 0 || j + k > height - 1) {
+
```
Comment (Himanshu40) [inline helpers.cpp:41]:
```suggestion
```

File: helpers.cpp [hunk 4]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+            // sums values of the pixel and 8 neighboring ones, skips iteration if it goes outside the pic
+            for (int k {-1}; k < 2; ++k) {
+            
+                if (j + k < 0 || j + k > height - 1) {
+                
+                    continue;
+                }
+
+                for (int h {-1}; h < 2; ++h) {
+
```
Comment (Himanshu40) [inline helpers.cpp:46]:
```suggestion
```

File: helpers.cpp [hunk 5]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+            
+                if (j + k < 0 || j + k > height - 1) {
+                
+                    continue;
+                }
+
+                for (int h {-1}; h < 2; ++h) {
+                
+                    if (i + h < 0 || i + h > width - 1) {
+
```
Comment (Himanshu40) [inline helpers.cpp:48]:
```suggestion
```

File: helpers.cpp [hunk 6]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+                    sumGreen += image[j + k][i + h].rgbtGreen;
+                    sumRed   += image[j + k][i + h].rgbtRed;
+                    ++count;
+                }
+            }
+
+            // averages the sum to make picture look blurrier
+            tempImg[j][i].rgbtBlue  = round(sumBlue / (float)count);
+            tempImg[j][i].rgbtGreen = round(sumGreen / (float)count);
+            tempImg[j][i].rgbtRed   = round(sumRed / (float)count);
```
Comment (Himanshu40) [inline helpers.cpp:62]:
```suggestion
            tempImg[j][i].rgbtBlue  = round(sumBlue / static_cast<float>(count));
            tempImg[j][i].rgbtGreen = round(sumGreen / static_cast<float>(count));
            tempImg[j][i].rgbtRed   = round(sumRed / static_cast<float>(count));
```

File: helpers.cpp [hunk 7]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+        }
+    }
+
+    // copies values from temporary table
+    for (int i{0}; i < width; ++i) {
+
```
Comment (Himanshu40) [inline helpers.cpp:68]:
```suggestion
```

File: helpers.cpp [hunk 8]
```diff
@@ -18,7 +18,61 @@ void reflect(const int &height, const int &width, RGBTRIPLE **image) {
...
+            tempImg[j][i].rgbtGreen = round(sumGreen / (float)count);
+            tempImg[j][i].rgbtRed   = round(sumRed / (float)count);
+        }
+    }
+
+    // copies values from temporary table
+    for (int i{0}; i < width; ++i) {
+    
+        for (int j{0}; j < height; ++j) {
+
```
Comment (Himanshu40) [inline helpers.cpp:70]:
```suggestion
```

**Review by Himanshu40 — CHANGES_REQUESTED** [review]
Please remove unnecessary lines.

**Review by Himanshu40 — APPROVED** [review]

**Pull request merged**